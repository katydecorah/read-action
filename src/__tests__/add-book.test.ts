import addBook from "../add-book";
import { promises, readFileSync } from "fs";
import book from "./fixture.json";
import { exportVariable } from "@actions/core";

const books = readFileSync("./_data/read.json", "utf-8");
const dateFinished = "2020-09-12";

jest.mock("@actions/core");

describe("addBook", () => {
  test("works", async () => {
    jest.spyOn(promises, "readFile").mockResolvedValueOnce(books);
    expect(
      await addBook(
        {
          dates: {
            dateAdded: undefined,
            dateStarted: undefined,
            dateFinished,
          },
          notes: "Amazing!",
          bookIsbn: "0525658181",
          providers: [],
          bookStatus: "finished",
        },
        book,
        "_data/read.yml"
      )
    ).toMatchSnapshot();
    expect(exportVariable).toHaveBeenNthCalledWith(
      1,
      "BookThumbOutput",
      "book-0525658181.png"
    );
    expect(exportVariable).toHaveBeenNthCalledWith(
      2,
      "BookThumb",
      "https://books.google.com/books/content?id=ty19yQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
    );
  });

  test("works, no images", async () => {
    jest.spyOn(promises, "readFile").mockResolvedValueOnce(books);
    expect(
      await addBook(
        {
          dates: {
            dateAdded: undefined,
            dateStarted: undefined,
            dateFinished,
          },
          notes: "Amazing!",
          bookIsbn: "0525658181",
          providers: [],
          bookStatus: "finished",
        },

        {
          ...book,
          imageLinks: {},
        },

        "_data/read.json"
      )
    ).toMatchSnapshot();
    expect(exportVariable).not.toHaveBeenNthCalledWith(
      1,
      "BookThumbOutput",
      "book-0525658181.png"
    );
    expect(exportVariable).not.toHaveBeenNthCalledWith(
      2,
      "BookThumb",
      "https://books.google.com/books/content?id=ty19yQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
    );
  });

  test("works, empty file", async () => {
    jest.spyOn(promises, "readFile").mockResolvedValueOnce("");
    expect(
      await addBook(
        {
          dates: {
            dateAdded: undefined,
            dateStarted: undefined,
            dateFinished,
          },
          notes: "Brilliant!",
          bookIsbn: "0525658181",
          providers: [],
          bookStatus: "finished",
        },

        book,
        "_data/read.json"
      )
    ).toMatchSnapshot();
  });
});
