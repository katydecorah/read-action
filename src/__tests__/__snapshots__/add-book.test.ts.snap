// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addBook works 1`] = `
[
  {
    "authors": [
      "Raven Leilani",
    ],
    "dateFinished": "2020-09-06",
    "description": "Sharp, comic, disruptive, tender, Raven Leilani's debut novel, Luster, sees a young black woman fall into art and someone else's open marriage. Edie is stumbling her way through her twenties--sharing a subpar apartment in Bushwick, clocking in and out of her admin job, making a series of inappropriate sexual choices. She's also, secretly, haltingly, figuring her way into life as an artist. And then she meets Eric, a digital archivist with a family in New Jersey, including an autopsist wife who has agreed to an open marriage--with rules. As if navigating the constantly shifting landscapes of contemporary sexual manners and racial politics weren't hard enough, Edie finds herself unemployed and falling into Eric's family life, his home. She becomes a hesitant friend to his wife and a de facto role model to his adopted daughter. Edie is the only black woman who young Akila knows. Razor sharp, darkly comic, sexually charged, socially disruptive, Luster is a portrait of a young woman trying to make her sense of her life in a tumultuous era. It is also a haunting, aching description of how hard it is to believe in your own talent and the unexpected influences that bring us into ourselves along the way.",
    "isbn": "9780385696005",
    "language": "en",
    "link": "https://books.google.com/books/about/Luster.html?hl=&id=eJ06zQEACAAJ",
    "pageCount": 240,
    "printType": "BOOK",
    "publishedDate": "2020-08-04",
    "thumbnail": "https://books.google.com/books/content?id=eJ06zQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
    "title": "Luster",
  },
  {
    "authors": [
      "Yaa Gyasi",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2020-09-12",
    "description": "A novel about faith, science, religion, and family that tells the deeply moving portrait of a family of Ghanaian immigrants ravaged by depression and addiction and grief, narrated by a fifth year candidate in neuroscience at Stanford school of medicine studying the neural circuits of reward seeking behavior in mice…",
    "isbn": "0525658181",
    "language": "en",
    "link": "https://books.google.com/books/about/Transcendent_Kingdom.html?hl=&id=ty19yQEACAAJ",
    "notes": "Amazing!",
    "pageCount": 288,
    "printType": "BOOK",
    "publishedDate": "2020",
    "thumbnail": "https://books.google.com/books/content?id=ty19yQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
    "title": "Transcendent Kingdom",
  },
  {
    "authors": [
      "Anna Wiener",
    ],
    "categories": [
      "Biography & Autobiography",
    ],
    "dateFinished": "2020-11-22",
    "description": "One of Vogue''s 22 Books to Read This Winter "A definitive document of a world in transition: I won''t be alone in returning to Uncanny Valley for clarity and consolation for many years to come." —Jia Tolentino, author of Trick Mirror: Reflections on Self-Delusion The prescient, page-turning account of a journey in Silicon Valley: a defining memoir of our digital age In her mid-twenties, at the height of tech industry idealism, Anna Wiener—stuck, broke, and looking for meaning in her work, like any good millennial--left a job in book publishing for the promise of the new digital economy. She moved from New York to San Francisco, where she landed at a big-data startup in the heart of the Silicon Valley bubble: a world of surreal extravagance, dubious success, and fresh-faced entrepreneurs hell-bent on domination, glory, and, of course, progress. Anna arrived amidst a massive cultural shift, as the tech industry rapidly transformed into a locus of wealth and power rivaling Wall Street. But amid the company ski vacations and in-office speakeasies, boyish camaraderie and ride-or-die corporate fealty, a new Silicon Valley began to emerge: one in far over its head, one that enriched itself at the expense of the idyllic future it claimed to be building. Part coming-age-story, part portrait of an already-bygone era, Anna Wiener’s memoir is a rare first-person glimpse into high-flying, reckless startup culture at a time of unchecked ambition, unregulated surveillance, wild fortune, and accelerating political power. With wit, candor, and heart, Anna deftly charts the tech industry’s shift from self-appointed world savior to democracy-endangering liability, alongside a personal narrative of aspiration, ambivalence, and disillusionment. Unsparing and incisive, Uncanny Valley is a cautionary tale, and a revelatory interrogation of a world reckoning with consequences its unwitting designers are only beginning to understand.",
    "isbn": "9780374719760",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=5fqTDwAAQBAJ",
    "pageCount": 288,
    "printType": "BOOK",
    "publishedDate": "2020-01-14",
    "thumbnail": "https://books.google.com/books/content?id=5fqTDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Uncanny Valley",
  },
  {
    "authors": [
      "Kimberly Drew",
      "Jenna Wortham",
    ],
    "categories": [
      "Social Science",
    ],
    "dateFinished": "2021-01-10",
    "description": "NEW YORK TIMES EDITORS’ CHOICE • An archive of collective memory and exuberant testimony A luminous map to navigate an opaque and disorienting present An infinite geography of possible futures What does it mean to be Black and alive right now? Kimberly Drew and Jenna Wortham have brought together this collection of work—images, photos, essays, memes, dialogues, recipes, tweets, poetry, and more—to tell the story of the radical, imaginative, provocative, and gorgeous world that Black creators are bringing forth today. The book presents a succession of startling and beautiful pieces that generate an entrancing rhythm: Readers will go from conversations with activists and academics to memes and Instagram posts, from powerful essays to dazzling paintings and insightful infographics. In answering the question of what it means to be Black and alive, Black Futures opens a prismatic vision of possibility for every reader.",
    "isbn": "9780399181146",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=pUGGDwAAQBAJ",
    "pageCount": 544,
    "printType": "BOOK",
    "publishedDate": "2020-12-01",
    "thumbnail": "https://books.google.com/books/content?id=pUGGDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Black Futures",
  },
  {
    "authors": [
      "Silvia Moreno-Garcia",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2021-09-26",
    "description": "NEW YORK TIMES BESTSELLER • “It’s Lovecraft meets the Brontës in Latin America, and after a slow-burn start Mexican Gothic gets seriously weird.”—The Guardian IN DEVELOPMENT AS A HULU ORIGINAL LIMITED SERIES PRODUCED BY KELLY RIPA AND MARK CONSUELOS • FINALIST FOR THE LOCUS AWARD • NOMINATED FOR THE BRAM STOKER AWARD • NAMED ONE OF THE BEST BOOKS OF THE YEAR BY The New Yorker • Vanity Fair • NPR • The Washington Post • Tordotcom • Marie Claire • Vox • Mashable • Men’s Health • Library Journal • Book Riot • LibraryReads An isolated mansion. A chillingly charismatic aristocrat. And a brave socialite drawn to expose their treacherous secrets. . . . From the author of Gods of Jade and Shadow comes “a terrifying twist on classic gothic horror” (Kirkus Reviews) set in glamorous 1950s Mexico. After receiving a frantic letter from her newly-wed cousin begging for someone to save her from a mysterious doom, Noemí Taboada heads to High Place, a distant house in the Mexican countryside. She’s not sure what she will find—her cousin’s husband, a handsome Englishman, is a stranger, and Noemí knows little about the region. Noemí is also an unlikely rescuer: She’s a glamorous debutante, and her chic gowns and perfect red lipstick are more suited for cocktail parties than amateur sleuthing. But she’s also tough and smart, with an indomitable will, and she is not afraid: Not of her cousin’s new husband, who is both menacing and alluring; not of his father, the ancient patriarch who seems to be fascinated by Noemí; and not even of the house itself, which begins to invade Noemi’s dreams with visions of blood and doom. Her only ally in this inhospitable abode is the family’s youngest son. Shy and gentle, he seems to want to help Noemí, but might also be hiding dark knowledge of his family’s past. For there are many secrets behind the walls of High Place. The family’s once colossal wealth and faded mining empire kept them from prying eyes, but as Noemí digs deeper she unearths stories of violence and madness. And Noemí, mesmerized by the terrifying yet seductive world of High Place, may soon find it impossible to ever leave this enigmatic house behind. “It’s as if a supernatural power compels us to turn the pages of the gripping Mexican Gothic.”—The Washington Post “Mexican Gothic is the perfect summer horror read, and marks Moreno-Garcia with her hypnotic and engaging prose as one of the genre’s most exciting talents.”—Nerdist “A period thriller as rich in suspense as it is in lush ’50s atmosphere.”—Entertainment Weekly",
    "isbn": "9780525620792",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=ksKyDwAAQBAJ",
    "pageCount": 320,
    "printType": "BOOK",
    "publishedDate": "2020-06-30",
    "thumbnail": "https://books.google.com/books/content?id=ksKyDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Mexican Gothic",
  },
  {
    "authors": [
      "bell hooks",
    ],
    "categories": [
      "Social Science",
    ],
    "dateFinished": "2022-01-13",
    "description": "A classic work of feminist scholarship, Ain't I a Woman has become a must-read for all those interested in the nature of black womanhood. Examining the impact of sexism on black women during slavery, the devaluation of black womanhood, black male sexism, racism among feminists, and the black woman's involvement with feminism, hooks attempts to move us beyond racist and sexist assumptions. The result is nothing short of groundbreaking, giving this book a critical place on every feminist scholar's bookshelf.",
    "isbn": "9781317588610",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=f2_fBQAAQBAJ",
    "pageCount": 206,
    "printType": "BOOK",
    "publishedDate": "2014-12-17",
    "thumbnail": "https://books.google.com/books/content?id=f2_fBQAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Ain't I a Woman",
  },
  {
    "authors": [
      "Kali Fajardo-Anstine",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2022-09-07",
    "description": "NATIONAL BESTSELLER • A dazzling epic of betrayal, love, and fate that spans five generations of an Indigenous Chicano family in the American West, from the author of the National Book Award finalist Sabrina & Corina A Phenomenal Book Club Pick • “Sometimes you just step into a book and let it wash over you, like you’re swimming under a big, sparkling night sky.”—Celeste Ng, author of Little Fires Everywhere and Everything I Never Told You ONE OF THE MOST ANTICIPATED BOOKS OF 2022—The Millions, Electric Lit, Lit Hub, Book Riot There is one every generation, a seer who keeps the stories. Luz “Little Light” Lopez, a tea leaf reader and laundress, is left to fend for herself after her older brother, Diego, a snake charmer and factory worker, is run out of town by a violent white mob. As Luz navigates 1930s Denver, she begins to have visions that transport her to her Indigenous homeland in the nearby Lost Territory. Luz recollects her ancestors’ origins, how her family flourished, and how they were threatened. She bears witness to the sinister forces that have devastated her people and their homelands for generations. In the end, it is up to Luz to save her family stories from disappearing into oblivion. Written in Kali Fajardo-Anstine’s singular voice, the wildly entertaining and complex lives of the Lopez family fill the pages of this multigenerational western saga. Woman of Light is a transfixing novel about survival, family secrets, and love—filled with an unforgettable cast of characters, all of whom are just as special, memorable, and complicated as our beloved heroine, Luz.",
    "isbn": "9780525511342",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=5LhBEAAAQBAJ",
    "pageCount": 336,
    "printType": "BOOK",
    "publishedDate": "2022-06-07",
    "thumbnail": "https://books.google.com/books/content?id=5LhBEAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Woman of Light",
  },
]
`;

exports[`addBook works, empty file 1`] = `
[
  {
    "authors": [
      "Yaa Gyasi",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2020-09-12",
    "description": "A novel about faith, science, religion, and family that tells the deeply moving portrait of a family of Ghanaian immigrants ravaged by depression and addiction and grief, narrated by a fifth year candidate in neuroscience at Stanford school of medicine studying the neural circuits of reward seeking behavior in mice…",
    "isbn": "0525658181",
    "language": "en",
    "link": "https://books.google.com/books/about/Transcendent_Kingdom.html?hl=&id=ty19yQEACAAJ",
    "notes": "Brilliant!",
    "pageCount": 288,
    "printType": "BOOK",
    "publishedDate": "2020",
    "thumbnail": "https://books.google.com/books/content?id=ty19yQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
    "title": "Transcendent Kingdom",
  },
]
`;

exports[`addBook works, no images 1`] = `
[
  {
    "authors": [
      "Raven Leilani",
    ],
    "dateFinished": "2020-09-06",
    "description": "Sharp, comic, disruptive, tender, Raven Leilani's debut novel, Luster, sees a young black woman fall into art and someone else's open marriage. Edie is stumbling her way through her twenties--sharing a subpar apartment in Bushwick, clocking in and out of her admin job, making a series of inappropriate sexual choices. She's also, secretly, haltingly, figuring her way into life as an artist. And then she meets Eric, a digital archivist with a family in New Jersey, including an autopsist wife who has agreed to an open marriage--with rules. As if navigating the constantly shifting landscapes of contemporary sexual manners and racial politics weren't hard enough, Edie finds herself unemployed and falling into Eric's family life, his home. She becomes a hesitant friend to his wife and a de facto role model to his adopted daughter. Edie is the only black woman who young Akila knows. Razor sharp, darkly comic, sexually charged, socially disruptive, Luster is a portrait of a young woman trying to make her sense of her life in a tumultuous era. It is also a haunting, aching description of how hard it is to believe in your own talent and the unexpected influences that bring us into ourselves along the way.",
    "isbn": "9780385696005",
    "language": "en",
    "link": "https://books.google.com/books/about/Luster.html?hl=&id=eJ06zQEACAAJ",
    "pageCount": 240,
    "printType": "BOOK",
    "publishedDate": "2020-08-04",
    "thumbnail": "https://books.google.com/books/content?id=eJ06zQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
    "title": "Luster",
  },
  {
    "authors": [
      "Yaa Gyasi",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2020-09-12",
    "description": "A novel about faith, science, religion, and family that tells the deeply moving portrait of a family of Ghanaian immigrants ravaged by depression and addiction and grief, narrated by a fifth year candidate in neuroscience at Stanford school of medicine studying the neural circuits of reward seeking behavior in mice…",
    "isbn": "0525658181",
    "language": "en",
    "link": "https://books.google.com/books/about/Transcendent_Kingdom.html?hl=&id=ty19yQEACAAJ",
    "notes": "Amazing!",
    "pageCount": 288,
    "printType": "BOOK",
    "publishedDate": "2020",
    "title": "Transcendent Kingdom",
  },
  {
    "authors": [
      "Anna Wiener",
    ],
    "categories": [
      "Biography & Autobiography",
    ],
    "dateFinished": "2020-11-22",
    "description": "One of Vogue''s 22 Books to Read This Winter "A definitive document of a world in transition: I won''t be alone in returning to Uncanny Valley for clarity and consolation for many years to come." —Jia Tolentino, author of Trick Mirror: Reflections on Self-Delusion The prescient, page-turning account of a journey in Silicon Valley: a defining memoir of our digital age In her mid-twenties, at the height of tech industry idealism, Anna Wiener—stuck, broke, and looking for meaning in her work, like any good millennial--left a job in book publishing for the promise of the new digital economy. She moved from New York to San Francisco, where she landed at a big-data startup in the heart of the Silicon Valley bubble: a world of surreal extravagance, dubious success, and fresh-faced entrepreneurs hell-bent on domination, glory, and, of course, progress. Anna arrived amidst a massive cultural shift, as the tech industry rapidly transformed into a locus of wealth and power rivaling Wall Street. But amid the company ski vacations and in-office speakeasies, boyish camaraderie and ride-or-die corporate fealty, a new Silicon Valley began to emerge: one in far over its head, one that enriched itself at the expense of the idyllic future it claimed to be building. Part coming-age-story, part portrait of an already-bygone era, Anna Wiener’s memoir is a rare first-person glimpse into high-flying, reckless startup culture at a time of unchecked ambition, unregulated surveillance, wild fortune, and accelerating political power. With wit, candor, and heart, Anna deftly charts the tech industry’s shift from self-appointed world savior to democracy-endangering liability, alongside a personal narrative of aspiration, ambivalence, and disillusionment. Unsparing and incisive, Uncanny Valley is a cautionary tale, and a revelatory interrogation of a world reckoning with consequences its unwitting designers are only beginning to understand.",
    "isbn": "9780374719760",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=5fqTDwAAQBAJ",
    "pageCount": 288,
    "printType": "BOOK",
    "publishedDate": "2020-01-14",
    "thumbnail": "https://books.google.com/books/content?id=5fqTDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Uncanny Valley",
  },
  {
    "authors": [
      "Kimberly Drew",
      "Jenna Wortham",
    ],
    "categories": [
      "Social Science",
    ],
    "dateFinished": "2021-01-10",
    "description": "NEW YORK TIMES EDITORS’ CHOICE • An archive of collective memory and exuberant testimony A luminous map to navigate an opaque and disorienting present An infinite geography of possible futures What does it mean to be Black and alive right now? Kimberly Drew and Jenna Wortham have brought together this collection of work—images, photos, essays, memes, dialogues, recipes, tweets, poetry, and more—to tell the story of the radical, imaginative, provocative, and gorgeous world that Black creators are bringing forth today. The book presents a succession of startling and beautiful pieces that generate an entrancing rhythm: Readers will go from conversations with activists and academics to memes and Instagram posts, from powerful essays to dazzling paintings and insightful infographics. In answering the question of what it means to be Black and alive, Black Futures opens a prismatic vision of possibility for every reader.",
    "isbn": "9780399181146",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=pUGGDwAAQBAJ",
    "pageCount": 544,
    "printType": "BOOK",
    "publishedDate": "2020-12-01",
    "thumbnail": "https://books.google.com/books/content?id=pUGGDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Black Futures",
  },
  {
    "authors": [
      "Silvia Moreno-Garcia",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2021-09-26",
    "description": "NEW YORK TIMES BESTSELLER • “It’s Lovecraft meets the Brontës in Latin America, and after a slow-burn start Mexican Gothic gets seriously weird.”—The Guardian IN DEVELOPMENT AS A HULU ORIGINAL LIMITED SERIES PRODUCED BY KELLY RIPA AND MARK CONSUELOS • FINALIST FOR THE LOCUS AWARD • NOMINATED FOR THE BRAM STOKER AWARD • NAMED ONE OF THE BEST BOOKS OF THE YEAR BY The New Yorker • Vanity Fair • NPR • The Washington Post • Tordotcom • Marie Claire • Vox • Mashable • Men’s Health • Library Journal • Book Riot • LibraryReads An isolated mansion. A chillingly charismatic aristocrat. And a brave socialite drawn to expose their treacherous secrets. . . . From the author of Gods of Jade and Shadow comes “a terrifying twist on classic gothic horror” (Kirkus Reviews) set in glamorous 1950s Mexico. After receiving a frantic letter from her newly-wed cousin begging for someone to save her from a mysterious doom, Noemí Taboada heads to High Place, a distant house in the Mexican countryside. She’s not sure what she will find—her cousin’s husband, a handsome Englishman, is a stranger, and Noemí knows little about the region. Noemí is also an unlikely rescuer: She’s a glamorous debutante, and her chic gowns and perfect red lipstick are more suited for cocktail parties than amateur sleuthing. But she’s also tough and smart, with an indomitable will, and she is not afraid: Not of her cousin’s new husband, who is both menacing and alluring; not of his father, the ancient patriarch who seems to be fascinated by Noemí; and not even of the house itself, which begins to invade Noemi’s dreams with visions of blood and doom. Her only ally in this inhospitable abode is the family’s youngest son. Shy and gentle, he seems to want to help Noemí, but might also be hiding dark knowledge of his family’s past. For there are many secrets behind the walls of High Place. The family’s once colossal wealth and faded mining empire kept them from prying eyes, but as Noemí digs deeper she unearths stories of violence and madness. And Noemí, mesmerized by the terrifying yet seductive world of High Place, may soon find it impossible to ever leave this enigmatic house behind. “It’s as if a supernatural power compels us to turn the pages of the gripping Mexican Gothic.”—The Washington Post “Mexican Gothic is the perfect summer horror read, and marks Moreno-Garcia with her hypnotic and engaging prose as one of the genre’s most exciting talents.”—Nerdist “A period thriller as rich in suspense as it is in lush ’50s atmosphere.”—Entertainment Weekly",
    "isbn": "9780525620792",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=ksKyDwAAQBAJ",
    "pageCount": 320,
    "printType": "BOOK",
    "publishedDate": "2020-06-30",
    "thumbnail": "https://books.google.com/books/content?id=ksKyDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Mexican Gothic",
  },
  {
    "authors": [
      "bell hooks",
    ],
    "categories": [
      "Social Science",
    ],
    "dateFinished": "2022-01-13",
    "description": "A classic work of feminist scholarship, Ain't I a Woman has become a must-read for all those interested in the nature of black womanhood. Examining the impact of sexism on black women during slavery, the devaluation of black womanhood, black male sexism, racism among feminists, and the black woman's involvement with feminism, hooks attempts to move us beyond racist and sexist assumptions. The result is nothing short of groundbreaking, giving this book a critical place on every feminist scholar's bookshelf.",
    "isbn": "9781317588610",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=f2_fBQAAQBAJ",
    "pageCount": 206,
    "printType": "BOOK",
    "publishedDate": "2014-12-17",
    "thumbnail": "https://books.google.com/books/content?id=f2_fBQAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Ain't I a Woman",
  },
  {
    "authors": [
      "Kali Fajardo-Anstine",
    ],
    "categories": [
      "Fiction",
    ],
    "dateFinished": "2022-09-07",
    "description": "NATIONAL BESTSELLER • A dazzling epic of betrayal, love, and fate that spans five generations of an Indigenous Chicano family in the American West, from the author of the National Book Award finalist Sabrina & Corina A Phenomenal Book Club Pick • “Sometimes you just step into a book and let it wash over you, like you’re swimming under a big, sparkling night sky.”—Celeste Ng, author of Little Fires Everywhere and Everything I Never Told You ONE OF THE MOST ANTICIPATED BOOKS OF 2022—The Millions, Electric Lit, Lit Hub, Book Riot There is one every generation, a seer who keeps the stories. Luz “Little Light” Lopez, a tea leaf reader and laundress, is left to fend for herself after her older brother, Diego, a snake charmer and factory worker, is run out of town by a violent white mob. As Luz navigates 1930s Denver, she begins to have visions that transport her to her Indigenous homeland in the nearby Lost Territory. Luz recollects her ancestors’ origins, how her family flourished, and how they were threatened. She bears witness to the sinister forces that have devastated her people and their homelands for generations. In the end, it is up to Luz to save her family stories from disappearing into oblivion. Written in Kali Fajardo-Anstine’s singular voice, the wildly entertaining and complex lives of the Lopez family fill the pages of this multigenerational western saga. Woman of Light is a transfixing novel about survival, family secrets, and love—filled with an unforgettable cast of characters, all of whom are just as special, memorable, and complicated as our beloved heroine, Luz.",
    "isbn": "9780525511342",
    "language": "en",
    "link": "https://play.google.com/store/books/details?id=5LhBEAAAQBAJ",
    "pageCount": 336,
    "printType": "BOOK",
    "publishedDate": "2022-06-07",
    "thumbnail": "https://books.google.com/books/content?id=5LhBEAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api",
    "title": "Woman of Light",
  },
]
`;
